### ISSUE TITLE
Use WebSocket from within worker?

### ISSUE THREAD

**Issue by FreeLiveTutor** [audreyt/node-webworker-threads#9]
Hi

Firstly, great work. I have a huge OCD for not using 3rd party libraries, but your 'node-webworker-threads' feels like it should be part of the core node.

As context, I've written a browser based wep-app that relies on WebWorkers, and now I'm able to use your library to simulate a headless client at the server side with node, with almost no coding changes to the App. A real breakthrough for me!

I've come up against a challenge though, as I also require WebSockets - _initiated from within the WebWorker_.

Inside my worker code I have : 
connection = new WebSocket( url, protocol); 

I've run some successfull tests from a node main thread using:
var WebSocket = require('websocket').client; [from https://github.com/Worlize/WebSocket-Node]

However I can't use require to pull in modules into a node-webworker.

Can you let me know your recommended approach to get around this?

Many thanks.

**Comment by saadnaji** [comment]
Hi audreyt,

Can you explain a little bit in detail in why WebSocket implementation using node JS , for example this one:

https://github.com/websockets/ws

does not work in webwoker threads?.

I am trying to use ws with  webwoker threads to debug and test some project that involve using these two. Although I am new NodeJS , I might offer to  help in implementing the missing stuff that web worker needs to get WebSocket needed for my own project if you explain a little bit. 

Thank you.

**Comment by audreyt** [comment]
Hi @saadnaji , the main issue is that, as described https://github.com/audreyt/node-webworker-threads/issues/9#issuecomment-17025886 workers run in a "pure" v8 Isolate, and does not have access to Node's `net` module, which the `ws` implementation would rely on.

**Comment by saadnaji** [comment]
Thank you for the reply,

I have knowledge in C programming which webwoker threads uses it in implementation. What is the best way to start contributing to help expose 'net' module?. Can you provide directions?. Like I said above, I started learning NodeJS last month and JS 3 months ago for  my undergraduate CS project.

**Comment by audreyt** [comment]
Yes. However, currently unless you need to pass very large amount of data between threads, or plan to spawn a huge amount of threads, the process-based https://www.npmjs.com/package/workerjs should already work with `ws` and is a more practical solution with the same API.

Similar discussions: https://github.com/iojs/io.js/issues/180 https://github.com/node-forward/discussions/issues/25
